components/myft-concept-button/src/js/myft-concept-button.js

```
@@ -0,0 +1,104 @@
+export default class MyftConceptButton {
+	// TODO make a @typedef for options
+	/**
+	 * Class constructor.
+	 *
+	 * @param {HTMLElement=} [element] The component element in the DOM
+	 * @param {object} [options={}] An options object for configuring the component
+	 * @property {HTMLElement} element The component element in the DOM
+	 * @property {object} options The merged options
+	 */
+	constructor(element, options) {
+		this.element = element;
+		this.button = this.element.querySelector('.myft-concept-button__button');
+		this.live = this.element.querySelector('.myft-follow-button__announcement');
+		this.options = Object.assign({}, options || this.getOptionsFromDom());
+	}
+
+	get isPressed() {
+		return this.button.ariaPressed === 'true';
+	}
+
+	/**
+	 * Set the followed state
+	 * Updates the UI and adds text to the a11y box.
+	 *
+	 * @param {boolean} [state] the state to set
+	 */
+	set isPressed(state) {
+		this.button.ariaPressed = state ? 'true' : 'false';
+
+		if (
+			this.options.ariaLivePressedText &&
```

question: for these three if statements - should we throw an error if only one of the options from each pair is supplied? I.E. If `this.options.ariaLivePressedText` is supplied but `this.options.ariaLiveUnpressedText` has not been supplied

---

yes this should probably error

---

also if they add the `ariaLivePressedText` but `this.live` doesn't exist maybe we should make it and inject it